### ISSUE TITLE
disable=redefined-builtin ignored in some scenarios

### ISSUE THREAD

**Issue by kfsone** [PyCQA/pylint#6042]
### Bug description

We encountered an issue where under some circumstances Pylint will complain about a redefined-builtin on a line with an explicit `# pylint: disable=redefined-builtin`, and/or W0622. _Only_ when the affected file is mentioned explicitly and not by search path (`pylint src` vs `pylint src/file.py`)

I was able to repro this on several of our linux ci workers, but it did not manifest on any of the windows or macos workers.

An MVCE is available here: https://github.com/kfsone/pylint-spurious-w0622

### Configuration

```ini
[MASTER]
extension-pkg-whitelist=
ignore=CVS,.svn,.git
ignore-patterns=
init-hook=sys.path.extend(["./Scripts", "../Scripts"])
jobs=1
load-plugins=pylint.extensions.emptystring,
			 pylint.extensions.bad_builtin,
			 pylint.extensions.docstyle,
			 pylint.extensions.overlapping_exceptions,
             pylint.extensions.check_elif,
             pylint.extensions.redefined_variable_type,
             pylint.extensions.typing,
persistent=yes
suggestion-mode=yes
unsafe-load-any-extension=no

[MESSAGES CONTROL]
confidence=
disable=
		docstring-first-line-empty,
        attribute-defined-outside-init, # eliminate
        bad-continuation,               # eliminate
        bad-whitespace,
        bare-except,
        broad-except,
        consider-using-f-string,        ###PY2-TO-3
        fixme,
        global-statement,
        invalid-name,
        keyword-arg-before-vararg,      # dangerous
        line-too-long,                  # eliminate
        logging-format-interpolation,   # eliminate
        missing-docstring,
        no-init,
        no-self-use,
        redefined-outer-name,           # dangerous
        too-few-public-methods,
        too-many-arguments,
        too-many-branches,
        too-many-instance-attributes,
        too-many-lines,
        too-many-locals,
        too-many-nested-blocks,         # eliminate
        too-many-public-methods,
        too-many-return-statements,     # eliminate
        too-many-statements,
        trailing-newlines,              # eliminate
        unspecified-encoding,           ###PY2-TO-3
        unused-argument,
        unused-import,                  # eliminate
        unused-wildcard-import,         # eliminate
        wildcard-import,                # eliminate
        wrong-import-order,
        wrong-import-position,
		duplicate-code,
		raise-missing-from,				# eliminate
		import-outside-toplevel,		# eliminate
		consider-using-in,				# eliminate
        deprecated-typing-alias,
        consider-using-alias,           # eliminate
		consider-alternative-union-syntax, # ELIMINATE
enable=c-extension-no-member

[REPORTS]
evaluation=10.0 - ((float(5 * error + warning + refactor + convention) / statement) * 10)
output-format=parseable
reports=no
score=yes

[REFACTORING]
max-nested-blocks=5
never-returning-functions=optparse.Values,sys.exit

[BASIC]
argument-naming-style=snake_case
attr-naming-style=snake_case
bad-names=foo,
          bar,
          baz,
          toto,
          tutu,
          tata
class-attribute-naming-style=any
class-naming-style=PascalCase
const-naming-style=UPPER_CASE
docstring-min-length=-1
function-naming-style=snake_case
good-names=i,
           j,
           _,
           logger
include-naming-hint=yes
inlinevar-naming-style=any
method-naming-style=snake_case
module-naming-style=snake_case
name-group=
no-docstring-rgx=^_
property-classes=abc.abstractproperty
variable-naming-style=snake_case

[FORMAT]
expected-line-ending-format=
ignore-long-lines=^(\s*(# )?<?https?://\S+>?|.*add_argument.*help=.*)$
indent-after-paren=4
indent-string='    '
max-line-length=120
max-module-lines=1000
no-space-check=
single-line-class-stmt=no
single-line-if-stmt=no

[LOGGING]
logging-modules=logging

[MISCELLANEOUS]
notes=FIXME,
      XXX,
      TODO

[SIMILARITIES]
ignore-comments=yes
ignore-docstrings=yes
ignore-imports=no
min-similarity-lines=4

[SPELLING]
max-spelling-suggestions=4
spelling-dict=
spelling-ignore-words=
spelling-private-dict-file=
spelling-store-unknown-words=no

[TYPECHECK]
contextmanager-decorators=contextlib.contextmanager
generated-members=
ignore-mixin-members=yes
ignore-on-opaque-inference=yes
ignored-classes=optparse.Values,thread._local,_thread._local
ignored-modules=
missing-member-hint=yes
missing-member-hint-distance=1
missing-member-max-choices=1

[VARIABLES]
additional-builtins=
allow-global-unused-variables=yes
callbacks=cb_,
          _cb
dummy-variables-rgx=_+$|(_[a-zA-Z0-9_]*[a-zA-Z0-9]+?$)|dummy|^ignored_|^unused_
ignored-argument-names=_.*|^ignored_|^unused_
init-import=no
redefining-builtins-modules=six.moves,past.builtins,future.builtins,io,builtins

[DESIGN]
max-args=5
max-attributes=7
max-bool-expr=5
max-branches=12
max-complexity=10
max-locals=15
max-parents=7
max-public-methods=20
max-returns=6
max-statements=50
min-public-methods=2
[CLASSES]
defining-attr-methods=__init__,
                      __new__,
                      setUp
exclude-protected=_asdict,
                  _fields,
                  _replace,
                  _source,
                  _make
valid-classmethod-first-arg=cls
valid-metaclass-classmethod-first-arg=mcs

[IMPORTS]
allow-wildcard-with-all=no
analyse-fallback-blocks=no
deprecated-modules=regsub,
                   TERMIOS,
                   Bastion,
                   rexec
ext-import-graph=
import-graph=
int-import-graph=
known-standard-library=
known-third-party=enchant

[PARAMETER_DOCUMENTATION]
accept-no-param-doc=yes
accept-no-raise-doc=yes
accept-no-return-doc=yes
accept-no-yields-doc=yes
default-docstring-type=default

[EXCEPTIONS]
overgeneral-exceptions=Exception

[TYPING]
runtime-typing=no
```

### Command used

```shell
# --persistent=no is optional
pylint --persistent=no --rcfile=pylintrc src
pylint --rcfile=pylintrc src/test_Testme.py src
pylint --rcfile=pylintrc src src/test_Testme.py

# but the problem did not occur when only specifying the files directly
pylint --rcfile=pylintrc src/test_testMe.py  # no error
pylint --rcfile=pylintrc src/*.py            # no error
```

### Pylint output

```shell
************* Module src.test_Testme
src/test_Testme.py:7: [W0622(redefined-builtin), setUp] Redefining built-in 'globals'

-----------------------------------
Your code has been rated at 8.33/10
```

### Expected behavior

The error should have been disabled, per the explicit `# pylint: disable=redefined-builtin`

### Pylint version

```shell
pylint 2.13.3
astroid 2.11.2
Python 3.8.10 (default, Mar 15 2022, 12:22:08)

pylint 2.12.2
astroid 2.9.3
Python 3.8.10 (default, Mar 15 2022, 12:22:08)
```

### OS / Environment

Ubuntu 20.04.3 LTS (GNU/Linux 5.11.0-1023-gcp x86_64)  -- reproducible
Windows 10 -- non-reproducible
MacOSX -- non-producible

**Comment by Pierre-Sassoulas** [comment]
Could be the same issue than #7006, the init-hook contains ``"``.

**Comment by jacobtylerwalls** [comment]
Ah, but the issue was opened on March 29, though.

**Comment by NathanEckert** [comment]
I encountered the same issue when bumping from pylint 2.12.2 to 2.15.10.

The issue happens for the following code:

```python
    def drop(
        self,
        *filter: Union[
            Condition[
                ColumnCoordinates,
                ConditionComparisonOperatorBound,
                Optional[Constant],
                ConditionCombinationOperatorBound,
            ],
            Mapping[str, Optional[ConstantValue]],
        ],  # pylint: disable=redefined-builtin
    ) -> None:
```

It is not reproducible on my machine (Ubuntu with Python 3.10.6), but is happening on our CI (CircleCI Docker machine, with Python 3.8.10)

**Comment by Pierre-Sassoulas** [comment]
It's often that the node you want to use message control is considered on another line when this kind of bug happen. To debug this you can use ``pylint a.py --enable=all``, you'll see the ``useless-suppression`` / ``suppressed-message`` / the actual message and be able to see the line where pylint expect you to disable:

```
************* Module a
a.py:11:0: I0011: Locally disabling redefined-builtin (W0622) (locally-disabled)
a.py:13:0: C0304: Final newline missing (missing-final-newline)
a.py:1:0: C0114: Missing module docstring (missing-module-docstring)
a.py:1:0: C0116: Missing function or method docstring (missing-function-docstring)
a.py:2:0: W0622: Redefining built-in 'filter' (redefined-builtin)
a.py:3:17: E0602: Undefined variable 'Union' (undefined-variable)
a.py:4:12: E0602: Undefined variable 'Condition' (undefined-variable)
a.py:5:16: E0602: Undefined variable 'ColumnCoordinates' (undefined-variable)
a.py:6:16: E0602: Undefined variable 'ConditionComparisonOperatorBound' (undefined-variable)
a.py:7:16: E0602: Undefined variable 'Optional' (undefined-variable)
a.py:7:25: E0602: Undefined variable 'Constant' (undefined-variable)
a.py:8:16: E0602: Undefined variable 'ConditionCombinationOperatorBound' (undefined-variable)
a.py:10:12: E0602: Undefined variable 'Mapping' (undefined-variable)
a.py:10:25: E0602: Undefined variable 'Optional' (undefined-variable)
a.py:10:34: E0602: Undefined variable 'ConstantValue' (undefined-variable)
a.py:2:8: W0613: Unused argument 'self' (unused-argument)
a.py:2:0: W0613: Unused argument 'filter' (unused-argument)
a.py:11:0: I0021: Useless suppression of 'redefined-builtin' (useless-suppression)
```
Here you disable line 11 but the message is emitted line 2 in a.py:**2**:0: W0622: Redefining built-in 'filter' (redefined-builtin)
